packages/mdc-textfield/foundation.js

```diff
@@ -167,4 +175,9 @@ export default class MDCTextfieldFoundation extends MDCFoundation {
       badInput: false,
     };
   }
+
+  setValid(isValid) {
+    this.useNativeValidityChecking_ = false;
```

Comment:
I'm curious of a use-case for that as well. I would think it's not typical (and potentially confusing) to change validation logic of a particular component instance on-the-fly, so you're either going to use this method directly to control valid state yourself, or you're going to rely on native validity checking.